crates/engine/tree/src/tree/mod.rs

```diff
@@ -766,31 +766,34 @@ where
             self.config.persistence_threshold()
     }
 
-    /// Returns a batch of consecutive canonical blocks to persist. The expected order is
-    /// oldest -> newest.
+    /// Returns a batch of consecutive canonical blocks to persist in the range
+    /// `(last_persisted_number .. canonical_head - threshold]` . The expected
+    /// order is oldest -> newest.
     fn get_canonical_blocks_to_persist(&self) -> Vec<ExecutedBlock> {
```

Comment:
could this return an empty vec?
if so we likely should add a sanity check before we send that vec to the persistence task

Reply:
sure, added here https://github.com/paradigmxyz/reth/commit/45fbe9edad2c86076653335885eecd462fad05c4